src/ol/sphere/sphere.js

```diff
@@ -58,6 +58,34 @@ ol.Sphere.prototype.cosineDistance = function(c1, c2) {
 
 
 /**
+ * Returns the area using Vincenty for a list of coordinates.
+ *
+ * [Reference](http://trs-new.jpl.nasa.gov/dspace/handle/2014/40409)
+ * Robert. G. Chamberlain and William H. Duquette, "Some Algorithms for
+ * Polygons on a Sphere", JPL Publication 07-03, Jet Propulsion
+ * Laboratory, Pasadena, CA, June 2007
+ *
+ * @param {Array.<ol.Coordinate>} coordinates List of coordinates.
```

Comment:
Should add that these are coordinates of a linear ring, and that the area will be positive if ring is oriented clockwise, otherwise it will be negative.